packages/api/tests/test-es-indexer.js

```diff
@@ -0,0 +1,552 @@
+'use strict';
+
+const test = require('ava');
+const sinon = require('sinon');
+const fs = require('fs');
+const path = require('path');
+const aws = require('@cumulus/common/aws');
+const { StepFunction } = require('@cumulus/ingest/aws');
+const { randomString } = require('@cumulus/common/test-utils');
+const indexer = require('../es/indexer');
+const { Search } = require('../es/search');
+const { bootstrapElasticSearch } = require('../lambdas/bootstrap');
+const granuleSuccess = require('./data/granule_success.json');
+const granuleFailure = require('./data/granule_failed.json');
+const pdrFailure = require('./data/pdr_failure.json');
+const pdrSuccess = require('./data/pdr_success.json');
+
+const esIndex = randomString();
+process.env.bucket = randomString();
+process.env.stackName = 'my-stack';
+process.env.ES_INDEX = esIndex;
+let esClient;
+
+
+test.before(async () => {
```

Comment:
Because all of these tests are using the same ES index and the same bucket, is there a chance that they could interfere with each other?  Could `test.beforeEach` be used?

Reply:
Because these tests rely on the `bucket` and `ES_INDEX` environment variables (which apply to everything in the process), and the tests run in parallel (https://www.npmjs.com/package/ava#process-isolation), there is not a way to run the tests in parallel using separate buckets and indexes.  For that reason, I think that

1. test.before should be test.beforeEach
2. The index and bucket should be random strings where process.env.XXX is set in the beforeEach block
3. Tests should be defined using `.serial`  (https://www.npmjs.com/package/ava#running-tests-serially)

Reply:
The tests are not interfering with each other right here. The `--serial` flag is also used in the `test` script.

In this particular case, the test is also more realistic because it adds multiple records to the same index.

In general, I agree with all the points you raised.

Reply:
`.serial()` and `--serial` do two different things.

`--serial` means "run each test *file* one at a time", and not specifying `--serial` means "run a separate node process for each test file, and run them all at once".

With or without the `--serial` flag, all of the tests in a given file are run in parallel.  If there are tests that must be run one at a time because they share some resource (like `process.env`, for instance), then those tests should be defined with `.serial()`.

(I finally got around to really digging into that part of the ava documentation this morning.)

I'm not going to push any harder on this, but I do think that these particular tests would be less likely to develop strange race condition issues in the future if they're run one at a time, and on fresh buckets / indexes.